docs/project/development-guidelines.md

```diff
@@ -0,0 +1,238 @@
+---
+id: development
+title: Development guides
+sidebar_label: Development guides
+slug: /project/development-guides
+custom_edit_url: null
+---
+
+## Defining the name of your usecase
+
+The name of the usecase must be something that refers to the business, making its real purpose clear to the business team.
+
+``` jsx
+
+// Well
+Search one or several funds
+Filter funds by custody
+Register user
+Update unit price
+Find the funds in Sinqia
+```
+
+``` jsx
+// bad
+list the information
+Calculates the percentage of the fund
+calculates the total
+Save user in bank
+Update PU
+Starts the textractClient and returns the JobId
+```
+
+Reference: [https://herbsjs.org/docs/usecase/features/#creating-a-use-case](https://herbsjs.org/docs/usecase/features/#creating-a-use-case)
+
+## **Creating custom methods in the repository**
+
+**Working with PostgreSQL**
+To create a query that comes out of CRUD, we can call knex directly, without the need to create a new variable and call it in the constructor.
+
+**Wrong:**
+
+By creating the variable this.conn and receiving the connection, we have already passed it to the mother class's knex.
+
+``` jsx
+// bad
+
+class UserRepository extends Repository {
+    constructor(injection) {
+        super({
+            entity: User,
+            table: "users",
+            knex: connection
+        })
+
+        this.conn = connection
+    }
+
+    async getUserByNickName(nickname) {
+        const ret = await this.conn('users').select('*')
+        .where('nickname', nickname)
+       
+        if (!ret[0] || ret[0].length == 0) return undefined
+        return User.fromJSON(ret[0])
+    }
+}
+
+```
+
+**Correct:**
+
+Call **this.knex** directly in our methods
+
+``` jsx
+// well
+
+class UserRepository extends Repository {
+    constructor(injection) {
+        super({
+            entity: User,
+            table: "users",
+            knex: connection
+        })
+    }
+
+    async getUserByNickName(nickname) {
+        const ret = await this.**knex**('users').select('*')
+         .where('nickname', nickname)
+
+        if (!ret[0] || ret[0].length == 0) return undefined
+        return User.fromJSON(ret[0])
+    }
+}
+
+```
+
+## Handling usecase failures
+
+To carry out the treatment of errors that occurred during the execution of the usecase, enable the environment variable HERBS_EXCEPTION in the production environments so that Herbs can obfuscate the errors generated and handle it on its return.
```

Comment:
## Handling Use Case failures

Herbs has two ways to handle errors that occur during the execution of a use case: (1) catch and audit and (2) let the error go through.

For the first option, you need to enable the environment variable [HERBS_EXCEPTION](https://herbsjs.org/docs/usecase/features#audit), best in the production environment, so that Herbs can catch the errors generated and audit them.

The second option is better for development environments, so developers can see the errors that occurred during the execution of the use case.

Additional to that, as a general best practice, you should only use try/catch if you know how to handle the error you caught. If you don’t know what to do with the error (which is 99% of the cases), let it go through and don’t use try/catch.